react/features/rtcstats/RTCStats.ts

```diff
@@ -47,9 +47,33 @@ function connectionFilter(config: any) {
  */
 class RTCStats {
     trace: any;
-    initialized = false;
+    options?: InitOptions;
+    isPeerConnectionWrapped = false;
     connStateEvents: any = [];
 
+    /**
+     * Initialize the rtcstats components, if the options have changed.
+     *
+     * @param {Oject} newOptions -.
+     * @returns {void}
+     */
+    maybeInit(newOptions: InitOptions) {
+        const oldOptions = this.options;
+        const changed = !oldOptions || (oldOptions.endpoint !== newOptions.endpoint
+            || oldOptions.meetingFqn !== newOptions.meetingFqn
+            || oldOptions.pollInterval !== newOptions.pollInterval
+            || oldOptions.sendSdp !== newOptions.sendSdp
+            || oldOptions.useLegacy !== newOptions.useLegacy);
+
+        if (changed) {
+            this.reset();
+
+            if (newOptions.meetingFqn && newOptions.endpoint) {
```

Comment:
I find this if condition a little suspicious. Is there a specific scenario you have in mind this should protect against?

What worries me here is that this could lead to the connection getting reset and not `init()` not getting called. Obviously calling `init()` without `meetingFqn` or `endpoint` will result in the WS connection not getting established properly. But the old, existing code didn't protected against that either.

Reply:
Makes sense. How about adding an warning or error for the else case of this then?
Just to make sure we notice if this should ever happen.

Reply:
Added a warning, that's a good point.